cli/compose/convert/service.go

```diff
@@ -510,9 +510,29 @@ func convertResources(source composetypes.Resources) (*swarm.ResourceRequirement
 				return nil, err
 			}
 		}
+
+		if len(source.Limits.GenericResources) != 0 {
+			return nil, fmt.Errorf("Unexpected Generic resources in Limits")
```

Comment:
This should be handled by the schema. It looks like the resource definition will need to be split up now that limits and reservations accept different values.